plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/CreateAppEngineStandardWtpProject.java

```diff
@@ -63,44 +49,31 @@ public void execute(IProgressMonitor monitor) throws InvocationTargetException,
     CreateProjectOperation operation = new CreateProjectOperation(
         description, "Creating new App Engine Project");
     try {
-      operation.execute(progress.newChild(20), uiInfoAdapter);
-      
-      IFacetedProject facetedProject = ProjectFacetsManager.create(
-          newProject, true, progress.newChild(20));
-      
-      installJavaFacet(facetedProject, progress.newChild(10));
-      
-      CodeTemplates.materialize(newProject, config, progress.newChild(20));
-      
-      installWebFacet(facetedProject, progress.newChild(10));
-      
-      // must happen after other two facets because the appengine facet requires them
-      FacetInstallDelegate.installAppEngineFacet(facetedProject, progress.newChild(10));
-      FacetInstallDelegate.installAppEngineRuntime(facetedProject, progress.newChild(20));
+      operation.execute(monitor, uiInfoAdapter);
+      CodeTemplates.materialize(newProject, config, monitor);
     } catch (ExecutionException ex) {
       throw new InvocationTargetException(ex, ex.getMessage());
-    } finally {
-      progress.done();
     }
-  }
 
-  static void installJavaFacet(IFacetedProject facetedProject, IProgressMonitor monitor) 
-      throws CoreException {
-    JavaFacetInstallConfig javaConfig = new JavaFacetInstallConfig();
-    List<IPath> sourcePaths = new ArrayList<>();
-    sourcePaths.add(new Path("src/main/java"));
-    sourcePaths.add(new Path("src/test/java"));
-    javaConfig.setSourceFolders(sourcePaths);
-    facetedProject.installProjectFacet(JavaFacet.VERSION_1_7, javaConfig, monitor);
-  }
+    Job facetInstallJob = new Job("Install App Engine Facet and runtimes in " + newProject.getName()) {
```

Comment:
Why does this have to happen as a job?  This `CreateAppEngineStandardWtpProject` is a `WorkspaceModifyOperation` and should be operating with a lock on the workspace.